### PR TITLE
Fix/student no supervisor

### PR DESCRIPTION
**Pull request by ErwanDerrien**

### CODE REVIEW

File: backend/src/main/java/com/team4/backend/controller/StudentController.java [hunk 1]
```diff
@@ -22,16 +24,17 @@
 public class StudentController {
 
     private final StudentService studentService;
+    private final SupervisorService supervisorService;
```
Comment (0xD34DC0DE) [inline backend/src/main/java/com/team4/backend/controller/StudentController.java:27]: add space in between line 26 and 27

File: backend/src/main/java/com/team4/backend/controller/SupervisorController.java [hunk 2]
```diff
@@ -52,9 +52,9 @@ public SupervisorController(SupervisorService supervisorService) {
         return supervisorService.getAllAssignedStudentsForCurrentSemester(supervisorId);
     }
 
-    @GetMapping("/{email}")
+    @GetMapping("/{getByEmail}")
```
Comment (0xD34DC0DE) [inline backend/src/main/java/com/team4/backend/controller/SupervisorController.java:55]: why did you change it to getByEmail? Email was fine.

File: backend/src/main/java/com/team4/backend/dto/UserDto.java [hunk 3]
```diff
@@ -19,4 +19,11 @@
 
     protected String profileImageId;
 
+
+    public UserDto(String id, String email, String firstName, String lastName) {
```
Comment (0xD34DC0DE) [inline backend/src/main/java/com/team4/backend/dto/UserDto.java:23]: Don't forget @Builder

File: frontend/src/components/contracts/CreateContractMonitorForm.js [hunk 4]
```diff
@@ -105,14 +105,12 @@ const CreateContractMonitorForm = ({ studentEmail, offerId }) => {
 
   return (
     <>
-      {/* TODO: find why must be == and not === so that code may work */}
```
Comment (0xD34DC0DE) [inline frontend/src/components/contracts/CreateContractMonitorForm.js:108]: lol

**Review by 0xD34DC0DE — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/team4/backend/controller/StudentController.java (hunk 1 above)
Comment (ErwanDerrien) [reply backend/src/main/java/com/team4/backend/controller/StudentController.java:27]: 👍

File: backend/src/main/java/com/team4/backend/controller/SupervisorController.java (hunk 2 above)
Comment (ErwanDerrien) [reply backend/src/main/java/com/team4/backend/controller/SupervisorController.java:55]: there was a todo saying it needed to be changed to follow the convention

File: backend/src/main/java/com/team4/backend/dto/UserDto.java (hunk 3 above)
Comment (ErwanDerrien) [reply backend/src/main/java/com/team4/backend/dto/UserDto.java:23]: 👍

File: frontend/src/components/contracts/CreateContractMonitorForm.js (hunk 4 above)
Comment (ErwanDerrien) [reply frontend/src/components/contracts/CreateContractMonitorForm.js:108]: tant pis pour les warnings, tant que ca marche

File: backend/src/main/java/com/team4/backend/controller/SupervisorController.java (hunk 2 above)
Comment (0xD34DC0DE) [reply backend/src/main/java/com/team4/backend/controller/SupervisorController.java:55]: 🤷‍♂️

**Review by 0xD34DC0DE — APPROVED** [review]
